### PR TITLE
Added Checklist to the PR Template and fixed minor grammar issues

### LINKED ISSUE #276: Add checklist to PR template
Example PR: https://github.com/scikit-image/scikit-image/pull/3983

### PR DESCRIPTION
**Pull request by saadhaxxan**
#### Reference Issues/PRs
Fixes #276 "Add a checklist to PR template"

#### What does this implement/fix? Explain your changes.
Added Checklist to the PR Template and fixed minor grammar issues

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -15,5 +15,29 @@ is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
...
+#### Checklist
+
+- [ ] Followed the [coding-guidelines](#guidelines).
+- [ ] Give your pull request a helpful title that summarizes what your contribution does.
+- [ ] Link your pull request to the issue (see: [closing keywords](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) for an easy way of linking your issue)
+- [ ] All public methods should have informative docstrings with sample usage presented as doctests when appropriate.
+- [ ] At least one paragraph of narrative documentation with links to references in the literature (with PDF links when possible) and the example.
+- [ ] If your feature is complex enough that a doctest is insufficient to fully showcase the utility, consider creating a Jupyter notebook to illustrate use instead.
+- [ ] All functions and classes must have unit tests. These should include, at the very least, type checking and ensuring correct computation/outputs.
+- [ ] All functions and classes should be rigorously typed with Python 3.5+
```
Comment (alyakin314) [inline .github/PULL_REQUEST_TEMPLATE.md:27]: this should 3.6

Comment (alyakin314) [reply .github/PULL_REQUEST_TEMPLATE.md:27]: be*

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -15,5 +15,29 @@ is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
...
+  ```bash
+  pip install black
+  black path/to/your_module.py
+  ```
+- [ ] Ensure all tests are passing locally using `pytest`. Install the necessary
+  packages by: 
+
+  ```bash
+  pip install pytest pytest-cov
+  pytest
```
Comment (alyakin314) [inline .github/PULL_REQUEST_TEMPLATE.md:39]: @bdpedigo
unrelated to this issue, but this breaks for me in different ways, depending on where i run it from.

if i run it from tests, i get #507 
if i run it from root of the repo, i get:
```
====================================================== ERRORS =======================================================
___________________________________________ ERROR collecting docs/conf.py ___________________________________________
docs/conf.py:93: in <module>
    import sphinx_rtd_theme
E   ModuleNotFoundError: No module named 'sphinx_rtd_theme'
============================================== short test summary info ==============================================
ERROR docs/conf.py - ModuleNotFoundError: No module named 'sphinx_rtd_theme'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================= 1 error in 3.84s ==================================================
```

Comment (alyakin314) [reply .github/PULL_REQUEST_TEMPLATE.md:39]: i see that we do recommend to install for devs it in readme

Comment (alyakin314) [reply .github/PULL_REQUEST_TEMPLATE.md:39]: installed and still break.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <!--
-Thanks for contributing a pull request! Please ensure you have taken a look at
+Thanks for contributing to a pull request! Please ensure you have taken a look at
```
[the remark this answers is not in the document]
Comment (saadhaxxan) [reply .github/PULL_REQUEST_TEMPLATE.md]: done 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -15,5 +15,29 @@ is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
 #### What does this implement/fix? Explain your changes.
 
 
+#### Checklist
+
+- [ ] Followed the [coding-guidelines](#guidelines).
+- [ ] Give your pull request a helpful title that summarizes what your contribution does.
```
[the remark this answers is not in the document]
Comment (saadhaxxan) [reply .github/PULL_REQUEST_TEMPLATE.md]: donee 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -15,5 +15,29 @@ is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
...
 
+#### Checklist
+
+- [ ] Followed the [coding-guidelines](#guidelines).
+- [ ] Pull request has a helpful title that summarizes what your contribution does.
+- [ ] Linked your pull request to the issue (see: [closing keywords](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) for an easy way of linking your issue)
+- [ ] All public methods have informative docstrings with sample usage presented as doctests when appropriate.
+- [ ] It has at least one paragraph of narrative documentation with links to references in the literature (with PDF links when possible) and the example.
+- [ ] If your feature is complex enough that a doctest is insufficient to fully showcase the utility, consider creating a Jupyter notebook to illustrate use instead.
+- [ ] All functions and classes have unit tests. These should include, at the very least, type checking and ensuring correct computation/outputs.
```
Comment (dwaynepryce) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: I think you should change this to:

`- [ ] All functions and classes have unit tests. Required tests include type checking, bounds validation, correct computation/outputs, and all conceivable edge cases.`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -15,5 +15,29 @@ is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
...
+- [ ] All functions and classes has been rigorously typed with Python 3.6+ 
+  [`typehinting`](https://docs.python.org/3/library/typing.html). Validate your typehinting by running `mypy ./graspy`
```
Comment (dwaynepryce) [inline .github/PULL_REQUEST_TEMPLATE.md:28]: `mypy ./graspologic`

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (dwaynepryce) [reply .github/PULL_REQUEST_TEMPLATE.md:39]: @alyakin314 - what's the break error you're getting?  I had issues running `make doc` on macOS as well.

**Comment by dwaynepryce** [comment]
Great start on this @saadhaxxan!  Please see my comments and make a few adjustments then I'm pretty happy with this.

**Review by dwaynepryce — CHANGES_REQUESTED** [review]
Please see my inline comments first, then we're good to go on this 👍

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (saadhaxxan) [reply .github/PULL_REQUEST_TEMPLATE.md]: done

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (saadhaxxan) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done

**Comment by saadhaxxan** [comment]
Is this able to merge now ??

**Comment by bdpedigo** [comment]
Hi @saadhaxxan - we have been having a lot of debate internally about what we want in a PR checklist, so I think we are going to end up replacing this with something soon. Sorry for the lack of communication. Please let us know if there's anything else you'd want to work on, we still appreciate the PR!
